### PR TITLE
Refactor usage of admin checks for uniformity

### PR DESCRIPTION
**Pull request by daxmc99**
Security sensitive code should be highly visible. This refactor
standardizes security checks with the 🚨 symbol.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/external_services.go [hunk 1]
```diff
@@ -63,15 +63,16 @@ func (*schemaResolver) UpdateExternalService(ctx context.Context, args *struct {
 		Config      *string
 	}
 }) (*externalServiceResolver, error) {
-	externalServiceID, err := unmarshalExternalServiceID(args.Input.ID)
-	if err != nil {
-		return nil, err
-	}
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/external_services.go:66]:
```suggestion
```

File: cmd/frontend/graphqlbackend/external_services.go [hunk 2]
```diff
@@ -63,15 +63,16 @@ func (*schemaResolver) UpdateExternalService(ctx context.Context, args *struct {
...
 
 	// 🚨 SECURITY: Only site admins are allowed to update the user.
 	if err := backend.CheckCurrentUserIsSiteAdmin(ctx); err != nil {
 		return nil, err
 	}
+
+	externalServiceID, err := unmarshalExternalServiceID(args.Input.ID)
+	if err != nil {
+		return nil, err
+	}
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/external_services.go:75]:
```suggestion
	}
	
```

File: cmd/frontend/graphqlbackend/org.go [hunk 3]
```diff
@@ -250,16 +250,18 @@ func (*schemaResolver) AddUserToOrganization(ctx context.Context, args *struct {
 	Organization graphql.ID
 	Username     string
 }) (*EmptyResponse, error) {
-	var orgID int32
-	if err := relay.UnmarshalSpec(args.Organization, &orgID); err != nil {
-		return nil, err
-	}
+
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/org.go:253]:
```suggestion
```

File: cmd/frontend/graphqlbackend/site_alerts.go [hunk 4]
```diff
@@ -79,6 +71,14 @@ func init() {
...
 		}
 
+		conf.ContributeWarning(func(c conf.Unified) (problems conf.Problems) {
+			if c.ExternalURL == "" {
+				problems = append(problems, conf.NewSiteProblem("`externalURL` is required to be set for many features of Sourcegraph to work correctly."))
+			} else if conf.DeployType() != conf.DeployDev && strings.HasPrefix(c.ExternalURL, "http://") {
+				problems = append(problems, conf.NewSiteProblem("Your connection is not private. We recommend [configuring Sourcegraph to use HTTPS/SSL](https://docs.sourcegraph.com/admin/nginx)"))
+			}
+			return problems
+		})
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/site_alerts.go:81]: Hmm, I don't think move this block down here will have desired outcome. It would add N (duplicated) checks that produces this warning every time the alert func is being called.

What is the reason of this change?

File: cmd/frontend/graphqlbackend/user.go [hunk 5]
```diff
@@ -323,6 +323,6 @@ func viewerCanChangeUsername(ctx context.Context, userID int32) bool {
 		return true
 	}
 	// 🚨 SECURITY: Only site admins are allowed to change a user's username when auth.enableUsernameChanges == false.
-	isSiteAdminErr := backend.CheckCurrentUserIsSiteAdmin(ctx)
-	return isSiteAdminErr == nil
+	err := backend.CheckCurrentUserIsSiteAdmin(ctx)
+	return err == nil
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/user.go:327]:
```suggestion
	return backend.CheckCurrentUserIsSiteAdmin(ctx) == nil
```

:trollface:

File: cmd/frontend/internal/app/debug.go [hunk 6]
```diff
@@ -92,6 +92,7 @@ func addGrafana(r *mux.Router) {
 // adminOnly is a HTTP middleware which only allows requests by admins.
 func adminOnly(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// TODO (Dax): Does this require 🚨 everywhere its called?
```
Comment (unknwon) [inline cmd/frontend/internal/app/debug.go:95]: Technically, yes.

File: cmd/frontend/graphqlbackend/site_alerts.go (hunk 4 above)
Comment (slimsag) [reply cmd/frontend/graphqlbackend/site_alerts.go:81]: I agree, I would move this back to the original location above. I suspect a typo/mistake

**Comment by slimsag** [comment]
Nice change!

File: cmd/frontend/graphqlbackend/site_alerts.go (hunk 4 above)
Comment (daxmc99) [reply cmd/frontend/graphqlbackend/site_alerts.go:81]: Yeah, this was a mistake. Moving it back.

File: cmd/frontend/internal/app/debug.go [hunk 7]
```diff
@@ -73,6 +73,7 @@ func addGrafana(r *mux.Router) {
 			addNoGrafanaHandler(r)
 		} else {
 			prefix := "/grafana"
+			// 🚨 SECURITY Only admins have access to Grafana dashboard
```
Comment (unknwon) [inline cmd/frontend/internal/app/debug.go:76]:
```suggestion
			// 🚨 SECURITY: Only admins have access to Grafana dashboard
```

Sorry, consistency 😄

File: cmd/frontend/internal/app/debugproxies/handler.go [hunk 8]
```diff
@@ -128,6 +128,7 @@ func displayNameFromEndpoint(ep Endpoint) string {
 // reverseProxyFromHost creates a reverse proxy from specified host with the path prefix that will be stripped from
 // request before it gets sent to the destination endpoint.
 func reverseProxyFromHost(host string, pathPrefix string) http.Handler {
+	// 🚨 SECURITY Only admins can create reverse proxies from host
```
Comment (unknwon) [inline cmd/frontend/internal/app/debugproxies/handler.go:131]:
```suggestion
	// 🚨 SECURITY: Only admins can create reverse proxies from host
```
